source/plugins/go/input/lib/applicationinsights.go

```diff
@@ -0,0 +1,339 @@
+	appInsightsEndpoint := os.Getenv(envAppEndpoint)
+	customProperties["WorkspaceCloud"] = getWorkspaceCloud()
+	isProxyConfigured := false
+	if proxyEndpoint != "" {
+		customProperties["IsProxyConfigured"] = "true"
+		isProxyConfigured = true
+		if isProxyCACertConfigured() {
+			customProperties["IsProxyCACertConfigured"] = "true"
+		}
+	} else {
+		customProperties["IsProxyConfigured"] = "false"
+		isProxyConfigured = false
+		if isIgnoreProxySettings() {
+			customProperties["IsProxyConfigurationIgnored"] = "true"
+		}
+	}
+
+	aadAuthMSIMode := os.Getenv(envAADMSIAuth)
+	if aadAuthMSIMode != "" && strings.EqualFold(aadAuthMSIMode, "true") {
+		customProperties["aadAuthMSIMode"] = "true"
+	} else {
+		customProperties["aadAuthMSIMode"] = "false"
+	}
+
+	addonResizerVPAEnabled := os.Getenv(envAddonResizer)
+	if addonResizerVPAEnabled != "" && strings.EqualFold(addonResizerVPAEnabled, "true") {
+		customProperties["addonResizerVPAEnabled"] = "true"
+	}
+
+	if encodedAppInsightsKey != "" {
+		decodedAppInsightsKey, err := base64.StdEncoding.DecodeString(encodedAppInsightsKey)
+		if err != nil {
+			aiLogger.Printf("Error decoding Application Insights key: %s", err.Error())
+		}
+
+		telemetryClientConfig := appinsights.NewTelemetryConfiguration(string(decodedAppInsightsKey))
+
+		if appInsightsEndpoint != "" {
+			aiLogger.Printf("Setting Application Insights endpoint to %s", appInsightsEndpoint)
+			telemetryClientConfig.EndpointUrl = appInsightsEndpoint
+		}
+
+		if isProxyConfigured {
+			proxyEndpointUrl, err := url.Parse(proxyEndpoint)
+			if err != nil {
+				aiLogger.Printf("Error parsing proxy endpoint: %s", err.Error())
+				telemetryClient = nil
+				return
+			}
+
+			transport := &http.Transport{
+				Proxy: http.ProxyURL(proxyEndpointUrl),
+			}
+			httpClient := &http.Client{
+				Transport: transport,
+			}
+			telemetryClientConfig.Client = httpClient
+		}
+		telemetryClient = appinsights.NewTelemetryClientFromConfig(telemetryClientConfig)
+	}
+
+	telemetryOffStr := os.Getenv("DISABLE_TELEMETRY")
+	if strings.EqualFold(telemetryOffStr, "true") {
+		telemetryClient.SetIsEnabled(false)
+	}
+
+}
+
+func getContainerRuntimeInfo() {
+	containerRuntime := os.Getenv(envContainerRT)
+	if containerRuntime != "" {
+		customProperties["DockerVersion"] = containerRuntime
+		// Not doing this for windows since docker is being deprecated soon and we dont want to bring in the socket dependency.
+		if !isWindows {
+			if strings.EqualFold(containerRuntime, "docker") {
+				cli, err := client.NewClientWithOpts(client.WithHost("unix:///var/run/host/docker.sock"), client.WithAPIVersionNegotiation())
+				if err != nil {
+					panic(err)
+				}
+				defer cli.Close()
+
+				// Get Docker version information
+				versionInfo, err := cli.ServerVersion(context.Background())
+				if err != nil {
+					panic(err)
+				}
+
+				if versionInfo.Version != "" {
+					customProperties["DockerVersion"] = versionInfo.Version
+				}
```

Comment:
we can remove the docker specific code since all the clusters are migrated to containerd.  I dont think we have any customers who are still with docker and supported k8s version.